service/api/cadence/client.go

```diff
@@ -113,6 +115,42 @@ func (t *cadenceClient) StartInterpreterWorkflow(ctx context.Context, options ap
 	return run.RunID, nil
 }
 
+func (t *cadenceClient) StartWaitForStateCompletionWorkflow(ctx context.Context, options api.StartWorkflowOptions, args ...interface{}) (runId string, err error) {
```

Comment:
This workflow should only need a workflowId as parameter. The ID reusepolicy can be just hardcoded to "DISALLOW_REUSE"

Reply:
Also make sure to handle this error in service.go for already started error